main/gameData.c

```diff
@@ -22,4 +19,58 @@
     gameData->level = 1;
     gameData->frameCount = 0;
     gameData->coins = 0;
-}
\ No newline at end of file
+    gameData->combo = 0;
+    gameData->comboTimer = 0;
+}
+
+void updateLedsHpMeter(entityManager_t *entityManager, gameData_t *gameData){
+    if(entityManager->playerEntity == NULL){
+        return;
+    }
+
+    uint8_t hp = entityManager->playerEntity->hp;
+    if(hp > 3){
+        hp = 3;
+    }
+
+    //HP meter led pairs:
```

Comment:
There are 8 LEDs, not 6. I'm not sure if this was intentional or not, but you could use `NUM_LEDS` for loops

Reply:
Nevermind, I see you use it later.